### DISCUSSION TITLE
helm-controller: Should uninstall remediations keepHistory by default?

### DISCUSSION THREAD

**Discussion by seaneagan** [fluxcd/flux2#232] [Q&A] [answered]
When deleting a HelmRelease it makes sense to delete the history by default. But for automated uninstall remediations it may make sense to keep it for debugging/recovery purposes? The main use case I think would be when using the uninstall remediation strategy for upgrades, if it doesn't recover on the subsequent install, maybe it'd be useful to know where you started before the (failed) upgrade/uninstall/install, and even be able to rollback to that manually.

If necessary this could be overridden via `disableKeepHistory=true` on `spec.install.remediation` and `spec.upgrade.remediation` (only relevant when `strategy=uninstall`).

We'd probably want to rename `spec.uninstall.keepHistory` to `spec.delete.keepHistory` since it'd then only be relevant for uninstalls due to deleting the HelmRelease CR.

**Comment by seaneagan** [comment, marked as answer]
This would require setting `--replace` on the subsequent install to avoid "cannot reuse a name that is still in use". We could require the user to set `spec.install.replace=true` or just automatically set it in this case as we could use the `Remediated` condition to determine that an uninstall was previously performed.

**Reply by hiddeco** [reply]
I think KISS outweighs the added value in this case, as adding this would make the list of automated behaviour we have to document even more extensive. If a user would like to enable such a functionality, they can simply configure `spec.install.replace=true` and `spec.uninstall.keepHistory=true`.
